source-opuswatch contribution from SebasZwinkels

SebasZwinkels opened the pull request:
## What

This PR adds a new connector for source OPUSWatch (source-opuswatch).

#### Contributor-provided connector description:

> ## Unlock Horticultural Insights with Airbyte and OPUSWatch

**What is OPUSWatch?**
OPUSWatch is a smart wearable solution developed by OPUS Solutions B.V., based in the Netherlands. This innovative smartwatch is specifically designed for the horticulture industry to digitize and streamline various operational processes within greenhouses and nurseries. It empowers growers to gain real-time insights into their labor, productivity, and overall operations. Employees use the OPUSWatch to easily record their tasks, track their work on specific crops, and even manage orders directly from their wrist. This data is then fed into a central system for analysis and decision-making.

Website: opuswatch.nl

**What does the OPUSWatch Airbyte Connector do?**
The Airbyte connector for OPUSWatch allows you to seamlessly extract valuable data from the OPUSWatch platform and load it into your data warehouse or lake. By connecting OPUSWatch to Airbyte, you can centralize your horticultural operational data with other business-critical information, enabling comprehensive analysis and reporting.

**With the OPUSWatch Airbyte connector, you can retrieve the following streams of data:**
* **Clients:** Information about your clients or customers.
* **Locations:** Details about the different locations or areas within your greenhouse or nursery.
* **Rows:** Data pertaining to the specific rows or sections where plants are cultivated.
* **Users:** Information about the individuals who have access to the OPUSWatch system.
* **Workers:** Details about your workforce, including their identification and potentially other relevant attributes.
* **Worker Groups:** Information about how your workers are organized into teams or groups.
* **Tasks:** Data on the various activities or jobs that are performed by workers.
* **Task Groups:** Information about how tasks are categorized or grouped together.
* **Labels:** Details about the different labels used for plants, products, or processes.
* **Varieties:** Information on the different plant varieties being cultivated.
* **Registrations Initial:** A snapshot of initial work registrations recorded by the OPUSWatch devices.
* **Registrations Incremental:** Updates and new work registrations recorded by the OPUSWatch devices since the last synchronization.
* **Sessions Initial:** A snapshot of initial work sessions or periods tracked by the OPUSWatch devices.
* **Sessions Incremental:** Updates and new work sessions or periods tracked by the OPUSWatch devices since the last synchronization.

By leveraging the OPUSWatch Airbyte connector, horticultural businesses can break down data silos and gain a holistic view of their operations. This enables data-driven decision-making for optimizing labor allocation, improving productivity, and ultimately enhancing the efficiency of their greenhouse or nursery operations.
#### The contributor provided the following description of the change:

> Renamed the initial date-field to start_date AND adding better description of what OPUS is.
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key.  |  |
| `start_date` | `string` | Start Date.  | 20250101 |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| client |  | No pagination | ✅ |  ❌  |
| locations |  | No pagination | ✅ |  ❌  |
| rows |  | No pagination | ✅ |  ❌  |
| users |  | No pagination | ✅ |  ❌  |
| workers |  | No pagination | ✅ |  ❌  |
| worker groups |  | No pagination | ✅ |  ❌  |
| tasks |  | No pagination | ✅ |  ❌  |
| task groups |  | No pagination | ✅ |  ❌  |
| labels |  | No pagination | ✅ |  ❌  |
| varieties |  | No pagination | ✅ |  ❌  |
| registrations initial |  | DefaultPaginator | ✅ |  ❌  |
| registrations incremental |  | No pagination | ✅ |  ❌  |
| sessions initial |  | DefaultPaginator | ✅ |  ❌  |
| sessions incremental |  | No pagination | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Consider how the connector handles pagination ([docs](https://docs.airbyte.com/connector-development/connector-builder-ui/pagination)).
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

<!-- DO NOT REMOVE: connector-builder-new-connector-contribution -->

SebasZwinkels committed:
Submission for source-opuswatch from Connector Builder

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=58115) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=58115) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=58115) it.</sub>

marcosmarxm commented on airbyte-integrations/connectors/source-opuswatch/manifest.yaml:
```diff
@@ -0,0 +1,1496 @@
+        order: 0
+        title: API Key
+        airbyte_secret: true
+      initial_data:
```
```suggestion
      start_date:
```
You can make this a datetime field and later apply a transformation to this specific format.

marcosmarxm commented on docs/integrations/sources/opuswatch.md:
```diff
@@ -0,0 +1,38 @@
+# OPUSWatch
+Connect to your OPUS data
```
Can you provide a better description of what OPUS is? What is the website and API documentation?

marcosmarxm reviewed (commented):
Some comments, it is missing the square svg icon as well.

marcosmarxm commented:
@SebasZwinkels need you to sign the CLA to continue the review process.

SebasZwinkels committed:
Merge branch 'airbytehq:master' into SebasZwinkels/builder-contribute/source-opuswatch

SebasZwinkels committed:
Submission for source-opuswatch from Connector Builder

SebasZwinkels committed:
Update icon.svg

marcosmarxm commented:
@SebasZwinkels need you to sign the CLA to accept the contribution.

SebasZwinkels commented:
Okay, I've made the following changes as requested:
- Signed the CLA.
- Added a square SVG icon.
- Updated the description of OPUS with more details, including the website.
- Changed the 'initial_data' field to 'start_date' in the manifest.yaml file.

Could you please confirm that these changes meet the requirements? I'm relatively new to contributing via Airbyte's pull request system, so I'd appreciate it if you could let me know if there are any other steps I need to take or if there's anything else I should be aware of. Thanks for your help!

marcosmarxm reviewed (approved):
Congrats @SebasZwinkels awesome contribution!
